sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -279,21 +308,23 @@ class DefaultCostModel(conf: RapidsConf) extends CostModel {
           // different CAST operations have different costs, so we allow these to be configured
           // based on the data types involved
```

Comment:
I think it would be a lot more accurate to recursively walk the expression tree and sum up the values instead of trying to match things at the top level.  That way `5 + foo` can have the correct cost compared to `5 + foo.bar`